e2e/libs/volume/crd_volume.py

```diff
@@ -0,0 +1,299 @@
+import time
+import warnings
+import datetime
+import logging
+
+from volume.abstract_volume import AbstractVolume
+from volume.rest_volume import RestVolume
+from kubernetes import client
+from kubernetes.client.rest import ApiException
+
+Ki = 2**10
+Mi = 2**20
+Gi = 2**30
+
+retry_count = 200
+retry_interval = 1
+
+
+class CRDVolume(AbstractVolume):
+
+    def __init__(self, node_exec):
+        self.obj_api = client.CustomObjectsApi()
+        self.node_exec = node_exec
+
+    def get(self, volume_name=""):
+        volume = self.obj_api.get_namespaced_custom_object(
+            group="longhorn.io",
+            version="v1beta2",
+            namespace="longhorn-system",
+            plural="volumes",
+            name=volume_name
+        )
+        return volume
+
+    def delete(self, volume_name):
+        if volume_name == "":
+            logging.info("deleting all volumes")
+        else:
+            logging.info(f"deleting the volume {volume_name}")
+
+        resp = self.get(volume_name)
+        if resp == "":
+            raise Exception("failed to get the volume")
+
+        volume_list = resp['items']
+        if len(volume_list) == 0:
+            logging.warning("cannot find the volume")
+            return
+
+        for volume in volume_list:
+            volume_name = volume['metadata']['name']
+            logging.debug(f"deleting volume {volume_name}")
+            self.obj_api.delete_namespaced_custom_object(
+                group="longhorn.io",
+                version="v1beta2",
+                namespace="longhorn-system",
+                plural="volumes",
+                name=volume_name
+            )
+        logging.info("finished deleting volume")
+
+    def create(self, volume_name, size, replica_count, volume_type):
+        if volume_name == "" or size == "" or replica_count == "" or volume_type == "":
+            raise Exception(f"empty parameters: volume_name={volume_name}, size={size}, replica_count={replica_count}, volume_type={volume_type}")
```

Comment:
I don't think we need to check for all arguments in most of the functions. We can probably assume they all have a value...